flink-table/flink-table-api-java-bridge/src/test/java/org/apache/flink/table/sinks/TableSinkFactoryTest.java

```diff
@@ -0,0 +1,114 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.sinks;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.functions.sink.SinkFunction;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.catalog.CatalogTableImpl;
+import org.apache.flink.table.catalog.ObjectIdentifier;
+import org.apache.flink.table.connector.sink.DynamicTableSink;
+import org.apache.flink.table.connector.sink.SinkFunctionProvider;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.types.Row;
+
+import org.junit.Test;
+
+import javax.annotation.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests for {@link BlackHoleTableSinkFactory} and {@link PrintTableSinkFactory}.
+ */
+public class TableSinkFactoryTest {
+
+	private static final TableSchema TEST_SCHEMA = TableSchema.builder()
+		.field("f0", DataTypes.STRING())
+		.field("f1", DataTypes.BIGINT())
+		.field("f2", DataTypes.BIGINT())
+		.build();
+
+	@Test
+	public void testPrint() throws Exception {
+		Map<String, String> properties = new HashMap<>();
+		properties.put("connector", "print");
+
+		DynamicTableSink sink = FactoryUtil.createTableSink(
+				null,
+				ObjectIdentifier.of("", "", ""),
+				new CatalogTableImpl(TEST_SCHEMA, properties, ""),
+				new Configuration(),
+				Thread.currentThread().getContextClassLoader());
+
+		assertTrue(sink instanceof PrintTableSinkFactory.PrintSink);
+		DynamicTableSink.Context context = new DynamicTableSink.Context() {
+
+			@Override
+			public boolean isBounded() {
+				return false;
+			}
+
+			@Override
+			public DynamicTableSink.DataStructureConverter createDataStructureConverter(
+					DataType consumedDataType) {
+				return new DynamicTableSink.DataStructureConverter() {
+
+					@Override
+					public void open(Context context) {
+					}
+
+					@Nullable
+					@Override
+					public Object toExternal(@Nullable Object o) {
+						RowData row = (RowData) o;
+						return o == null ? null : Row.of(row.getInt(0), row.getString(1).toString());
```

Comment:
A way to verify print sink is delegating `System.out` to an `OutputStream`, then we can verify the outputs in the `OutputStream`. You can take `PrintSinkFunctionTest` as an example.